### PR TITLE
handlebars: Split text content on multiple lines

### PR DESCRIPTION
**Pull request by dcyriller**
## Description

```handlebars
{{!-- Input --}}
<div>
  A long enough string to trigger a line break that would prevent wrapping more and more.
</div>

{{!-- Prettier stable --}}
<div>
  A long enough string to trigger a line break that would prevent wrapping more and more.
</div>

{{!-- Prettier main --}}
<div>
  A long enough string to trigger a line break that would prevent wrapping more
  and more.
</div>
```

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-handlebars/clean.js [hunk 1]
```diff
@@ -1,20 +1,17 @@
...
 function clean(ast, newNode /*, parent*/) {
-  // (Glimmer/HTML) ignore TextNode whitespace
+  // (Glimmer/HTML) ignore TextNode
   if (ast.type === "TextNode") {
-    const trimmed = ast.chars.trim();
-    if (!trimmed) {
-      return null;
-    }
-    newNode.chars = trimmed;
+    return null;
```
Comment (fisker) [inline src/language-handlebars/clean.js:6]:
```suggestion
    const trimmed = ast.chars.trim();
    if (!trimmed) {
      return null;
    }
    newNode.chars = trimmed.replace(/[\t\n\f\r ]+/g, '');
```

Comment (fisker) [reply src/language-handlebars/clean.js:6]: We only change these characters, let's replace it with a ` `.

File: src/language-handlebars/printer-glimmer.js [hunk 2]
```diff
@@ -345,9 +356,11 @@ function print(path, options, print) {
         trailingSpace = "";
       }
 
+      text = text.replace(/^\s+/g, leadingSpace).replace(/\s+$/, trailingSpace);
```
Comment (fisker) [inline src/language-handlebars/printer-glimmer.js:359]: As you're on this, let's change `\s` to html whitespace?

File: src/language-handlebars/printer-glimmer.js [hunk 3]
```diff
@@ -356,7 +356,9 @@ function print(path, options, print) {
         trailingSpace = "";
       }
 
-      text = text.replace(/^\s+/g, leadingSpace).replace(/\s+$/, trailingSpace);
+      text = text
+        .replace(/^[\t\n\f\r ]+/g, leadingSpace)
```
Comment (fisker) [inline src/language-handlebars/printer-glimmer.js:360]: Let's see  if any test fails, if not, please add some test.

Comment (dcyriller) [reply src/language-handlebars/printer-glimmer.js:360]: I modified the git history to:
- add a test, I pushed this commit to the repo, it should ✅
- modify the printer to only replace html whitespaces (and not vertical tabulations), as a result CI should ❌
- record the tests, CI ✅

**Comment by dcyriller** [comment]
This one should be ready to review

**Review by fisker — APPROVED** [review]

**Review by thorn0 — APPROVED** [review]

**Pull request merged**